docs/source/en/tasks/prompting.md

```diff
@@ -0,0 +1,442 @@
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+
+# LLM prompting guide
+
+[[open-in-colab]]
+
+Large Language Models such as Falcon, LLaMA, etc. are pretrained transformer models initially trained to predict the 
+next token given some input text. They typically have billions of parameters and have been trained on trillions of 
+tokens for an extended period of time. As a result, these models become quite powerful and versatile, and you can use 
+them to solve multiple NLP tasks out of the box by instructing the models with natural language prompts.
+
+Designing such prompts to ensure the optimal output is often called "prompt engineering". Prompt engineering is an 
+iterative process that requires a fair amount of experimentation. Natural languages are much more flexible and expressive 
+than programming languages, however, they can also introduce some ambiguity. At the same time, prompts in natural language 
+are quite sensitive to changes. Even minor modifications in prompts can lead to wildly different outputs.
+
+While there is no exact recipe for creating prompts to match all cases, researchers have worked out a number of best 
+practices that help to achieve optimal results more consistently. 
+
+This guide covers the prompt engineering best practices to help you craft better LLM prompts and solve various NLP tasks. 
+You'll learn:
+
+- [Basics of prompting](#basic-prompts)
+- [Best practices of LLM prompting](#best-practices-of-llm-prompting)
+- [Advanced prompting techniques: few-shot prompting and chain-of-thought](#advanced-prompting-techniques)
+- [When to fine-tune instead of prompting](#prompting-vs-fine-tuning)
+
+<Tip>
+
+Prompt engineering is only a part of the LLM output optimization process. Another essential component is choosing the 
+optimal text generation strategy. You can customize how your LLM selects each of the subsequent tokens when generating 
+the text without modifying any of the trainable parameters. By tweaking the text generation parameters, you can reduce 
+repetition in the generated text and make it more coherent and human-sounding. 
+Text generation strategies and parameters are out of scope for this guide, but you can learn more about these topics in 
+the following guides: 
+ 
+* [Generation with LLMs](../llm_tutorial)
+* [Text generation strategies](../generation_strategies)
+
+</Tip>
+
+## Basics of prompting
+
+### Types of models 
+
+The majority of modern LLMs are decoder-only transformers. Some examples include: [LLaMA](../model_doc/llama), 
+[Llama2](../model_doc/llama2), [Falcon](../model_doc/falcon), [GPT2](../model_doc/gpt2). However, you may encounter  
+encoder-decoder transformer LLMs as well, for instance, [Flan-T5](../model_doc/flan-t5) and [BART](../model_doc/bart).
```

Comment:
```suggestion
[Llama2](../model_doc/llama2), [Falcon](../model_doc/falcon), [GPT2](../model_doc/gpt2). However, you may encounter
encoder-decoder transformer LLMs as well, for instance, [Flan-T5](../model_doc/flan-t5) and [BART](../model_doc/bart).
```

The additional space seems to add an unnecessary line return

![image](https://github.com/huggingface/transformers/assets/30755778/68a1255a-8908-4312-a73d-cadf1b7db26b)